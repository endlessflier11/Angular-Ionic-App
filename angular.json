{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "defaultProject": "app",
  "newProjectRoot": "projects",
  "projects": {
    "app": {
      "root": "",
      "sourceRoot": "src",
      "projectType": "application",
      "prefix": "csaa",
      "schematics": {},
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "www",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "src/assets",
                "output": "assets"
              },
              {
                "glob": "**/*",
                "input": "src/app/micro-apps/csaa-mobile/assets",
                "output": "assets/csaa-mobile"
              },
              {
                "glob": "**/*.svg",
                "input": "node_modules/ionicons/dist/ionicons/svg",
                "output": "./svg"
              }
            ],
            "styles": [
              {
                "input": "src/theme/variables.scss"
              },
              {
                "input": "src/global.scss"
              }
            ],
            "scripts": [],
            "allowedCommonJsDependencies": [
              "us-bank-account-validator",
              "card-validator",
              "crypto-js"
            ],
            "buildOptimizer": false,
            "sourceMap": true,
            "optimization": false
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                }
              ]
            },
            "ds": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.ds.ts"
                }
              ]
            },
            "development": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.ds.ts"
                }
              ]
            },
            "ci": {
              "progress": false
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "app:build",
            "proxyConfig": "proxy.conf.js"
          },
          "configurations": {
            "production": {
              "browserTarget": "app:build:production",
              "proxyConfig": "proxy.prod.conf.json"
            },
            "development": {
              "browserTarget": "app:build:development",
              "proxyConfig": "proxy.conf.json"
            },
            "ds": {
              "browserTarget": "app:build:ds",
              "proxyConfig": "proxy.conf.json"
            },
            "ci": {
              "progress": false
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "app:build"
          }
        },
        "test": {
          "builder": "@angular-builders/jest:run",
          "options": {
            "tsConfig": "tsconfig.spec.json"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
          }
        },
        "e2e": {
          "builder": "@angular-devkit/build-angular:protractor",
          "options": {
            "protractorConfig": "e2e/protractor.conf.js",
            "devServerTarget": "app:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "app:serve:production"
            },
            "ci": {
              "devServerTarget": "app:serve:ci"
            }
          }
        },
        "ionic-cordova-build": {
          "builder": "@ionic/angular-toolkit:cordova-build",
          "options": {
            "browserTarget": "app:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "app:build:production"
            }
          }
        },
        "ionic-cordova-serve": {
          "builder": "@ionic/angular-toolkit:cordova-serve",
          "options": {
            "cordovaBuildTarget": "app:ionic-cordova-build",
            "devServerTarget": "app:serve"
          },
          "configurations": {
            "production": {
              "cordovaBuildTarget": "app:ionic-cordova-build:production",
              "devServerTarget": "app:serve:production"
            }
          }
        }
      }
    }
  },
  "cli": {
    "defaultCollection": "@angular-eslint/schematics",
    "analytics": false
  },
  "schematics": {
    "@ionic/angular-toolkit:component": {
      "styleext": "scss"
    },
    "@ionic/angular-toolkit:page": {
      "styleext": "scss"
    }
  }
}
