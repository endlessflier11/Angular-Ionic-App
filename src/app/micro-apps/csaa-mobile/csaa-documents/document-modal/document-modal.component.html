<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot='start'>
      <csaa-back-button (back)='backButtonClick()'></csaa-back-button>
    </ion-buttons>
    <ion-title>Filter</ion-title>
    <ion-buttons slot='end'>
      <ion-button (click)='dismissModal()'>Done</ion-button>
    </ion-buttons>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ion-list>
    <ion-radio-group [value]='activeFilter' (ionChange)='onFilterChange($event)'>
      <ion-item>
        <ion-label>All Documents</ion-label>
        <ion-radio slot='end' mode='md' value='all_documents'></ion-radio>
      </ion-item>
      <ion-item *ngIf='checkPolicyTypes()'>
        <ion-label>Policy Documents</ion-label>
        <ion-radio slot='end' mode='md' value='policy_documents'></ion-radio>
      </ion-item>
      <ion-item *ngIf='checkBillingPayments()'>
        <ion-label>Billing & Payments</ion-label>
        <ion-radio slot='end' mode='md' value='billing_&_payments'></ion-radio>
      </ion-item>
      <ion-item *ngFor='let filterType of currentFilter'>
        <ion-label>{{ PolicyDocumentType[filterType] }}</ion-label>
        <ion-radio slot='end' mode='md' [value]='filterType'></ion-radio>
      </ion-item>
    </ion-radio-group>
  </ion-list>
</ion-content>
