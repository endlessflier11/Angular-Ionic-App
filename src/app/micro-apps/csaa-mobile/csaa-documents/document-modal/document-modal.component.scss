@import '../../_scss/variables';

ion-content {
  --background: var(--ion-color-primary-contrast);
}

