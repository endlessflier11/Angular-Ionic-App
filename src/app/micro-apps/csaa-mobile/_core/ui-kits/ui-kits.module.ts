import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { ThemeIconComponent } from './theme-icon/theme-icon.component';
import { WorkingHoursComponent } from './working-hours/working-hours.component';
import { MyPolicyCardComponent } from './my-policy-card/my-policy-card.component';
import { SkeletonCardComponent } from './skeleton-card/skeleton-card.component';
import { BackButtonComponent } from './back-button/back-button.component';
import { PhonePipe } from './pipes/phone-number.pipe';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card/csaa-auto-id-card.component';
import { CsaaLoadingComponent } from './csaa-loading/csaa-loading.component';
import { ContentRendererComponent } from './content-renderer/content-renderer.component';
import { NationalCustomAlertModalComponent } from './national-custom-alert/national-custom-alert-modal';
import { SafeDataPipe } from './pipes/safe-data/safe-data.pipe';
import { DocumentActionModalComponent } from './document-action-modal/document-action-modal.component';
import { CsaaPolicySubtitlesComponent } from './csaa-policy-subtitles/csaa-policy-subtitles.component';
import { PaperlessTermsComponent } from './paperless-terms/paperless-terms.component';
import { PaperlessTermsModalComponent } from './paperless-terms-modal/paperless-terms-modal.component';
import { AppleWalletButtonComponent } from './apple-wallet-button/apple-wallet-button.component';

const UI_COMPONENTS = [
  ThemeIconComponent,
  WorkingHoursComponent,
  MyPolicyCardComponent,
  SkeletonCardComponent,
  BackButtonComponent,
  CsaaAutoIdCardComponent,
  CsaaLoadingComponent,
  ContentRendererComponent,
  NationalCustomAlertModalComponent,
  DocumentActionModalComponent,
  CsaaPolicySubtitlesComponent,
  PaperlessTermsComponent,
  PaperlessTermsModalComponent,
  AppleWalletButtonComponent,
];

const UI_PIPES = [PhonePipe, SafeDataPipe];

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, IonicModule],
  declarations: [...UI_COMPONENTS, ...UI_PIPES],
  exports: [...UI_COMPONENTS, ...UI_PIPES],
  providers: [DatePipe],
})
export class UiKitsModule {}
