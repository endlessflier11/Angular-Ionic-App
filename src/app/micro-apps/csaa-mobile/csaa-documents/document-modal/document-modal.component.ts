import { Component, Input, OnInit } from '@angular/core';

import { ModalController } from '@ionic/angular';

import { PolicyType } from '../../_core/interfaces';
import {
  BILLING_AND_PAYMENTS_GROUP,
  POLICY_DOCUMENTS_GROUP,
  PolicyDocumentType,
} from '../../_core/interfaces/document.interface';
import { ConfigService } from '../../_core/services/config/config.service';
import { noop } from '../../_core/helpers';

@Component({
  selector: 'csaa-document-modal',
  templateUrl: './document-modal.component.html',
  styleUrls: ['./document-modal.component.scss'],
})
export class DocumentModalComponent implements OnInit {
  @Input() currentFilter: string[] = [];
  @Input() policyType: PolicyType;
  @Input() activeFilter: string;

  currentTheme = '';
  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyDocumentType = PolicyDocumentType;
  hasPolicyTypes = null;
  hasBillingPaymentTypes = null;

  constructor(private modalCtrl: ModalController, private configService: ConfigService) {}

  ngOnInit() {
    this.currentTheme = this.configService.getTheme();
  }

  checkPolicyTypes() {
    if (this.hasPolicyTypes === null) {
      this.currentFilter.forEach((f) => {
        this.hasPolicyTypes =
          this.hasPolicyTypes || POLICY_DOCUMENTS_GROUP.indexOf(PolicyDocumentType[f]) >= 0;
      });
    }
    return this.hasPolicyTypes;
  }

  checkBillingPayments() {
    if (this.hasBillingPaymentTypes === null) {
      this.currentFilter.forEach((f) => {
        this.hasBillingPaymentTypes =
          this.hasBillingPaymentTypes ||
          BILLING_AND_PAYMENTS_GROUP.indexOf(PolicyDocumentType[f]) >= 0;
      });
    }
    return this.hasBillingPaymentTypes;
  }

  onFilterChange(event) {
    this.activeFilter = event?.detail?.value;
  }

  dismissModal() {
    this.modalCtrl.dismiss(this.activeFilter).then(noop);
  }

  backButtonClick() {
    this.modalCtrl.dismiss(null).then(noop);
  }
}
