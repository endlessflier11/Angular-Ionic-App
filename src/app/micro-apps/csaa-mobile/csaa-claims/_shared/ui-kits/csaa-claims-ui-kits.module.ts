import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { AgentCardComponent } from './agent-card/agent-card.component';
import { ClaimsHistoryCardComponent } from './claims-history-card/claims-history-card.component';
import { ClaimsInfoCardComponent } from './claims-info-card/claims-info-card.component';
import { WhatToDoCardComponent } from './what-to-do-card/what-to-do-card.component';

const SHARED_COMPONENTS = [
  AgentCardComponent,
  ClaimsHistoryCardComponent,
  ClaimsInfoCardComponent,
  WhatToDoCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaClaimsUiKitsModule {}
