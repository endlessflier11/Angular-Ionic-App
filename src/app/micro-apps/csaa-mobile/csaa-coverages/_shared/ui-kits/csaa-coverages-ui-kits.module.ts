import { NgModule } from '@angular/core';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card/csaa-pup-coverages-card.component';
import { CsaaGeneralCoveragesCardComponent } from './csaa-general-coverages-card/csaa-general-coverages-card.component';
import { CsaaDriversCoveragesCardComponent } from './csaa-drivers-coverages-card/csaa-drivers-coverages-card.component';
import { CsaaHomeCoveragesCardComponent } from './csaa-home-coverages-card/csaa-home-coverages-card.component';
import { CsaaVehiclesCoveragesCardComponent } from './csaa-vehicles-coverages-card/csaa-vehicles-coverages-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaCoverageCardItemComponent } from './csaa-coverage-card-item/csaa-coverage-card-item.component';
import {
  CsaaCoveragesVehicleCoveredCardComponent
} from './csaa-vehicle-coverages-covered-card/csaa-coverages-vehicle-covered-card.component';
import {
  CsaaCoveragesVehicleNotCoveredCardComponent
} from './csaa-vehicle-coverages-not-covered-card/csaa-coverages-vehicle-not-covered-card.component';
import { CsaaDriversCoveragesContentV1Component } from './csaa-drivers-coverages-content-v1/csaa-drivers-coverages-content-v1.component';
import { CsaaDriversCoveragesContentV2Component } from './csaa-drivers-coverages-content-v2/csaa-drivers-coverages-content-v2.component';

const SHARED_COMPONENTS = [
  CsaaCoverageCardItemComponent,
  CsaaPupCoveragesCardComponent,
  CsaaGeneralCoveragesCardComponent,
  CsaaDriversCoveragesCardComponent,
  CsaaDriversCoveragesContentV1Component,
  CsaaHomeCoveragesCardComponent,
  CsaaVehiclesCoveragesCardComponent,
  CsaaCoveragesVehicleCoveredCardComponent,
  CsaaCoveragesVehicleNotCoveredCardComponent,
  CsaaDriversCoveragesContentV2Component,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaCoveragesUiKitsModule {}
