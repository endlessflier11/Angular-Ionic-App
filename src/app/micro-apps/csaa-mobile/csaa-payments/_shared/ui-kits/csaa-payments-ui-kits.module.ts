import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
