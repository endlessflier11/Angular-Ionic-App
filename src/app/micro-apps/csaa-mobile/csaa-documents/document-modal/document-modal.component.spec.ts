import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule, ModalController } from '@ionic/angular';

import { DocumentModalComponent } from './document-modal.component';
import { By, PageTestingModule } from '@app/testing';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { noop } from 'rxjs';

describe('DocumentModalComponent', () => {
  let component: DocumentModalComponent;
  let fixture: ComponentFixture<DocumentModalComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [DocumentModalComponent],
      imports: [
        PageTestingModule.withConfig({
          providesConfig: true,
          providesModal: true,
          providesAnalytics: true,
        }),
        IonicModule,
        UiKitsModule,
      ],
    })
      .compileComponents()
      .then(noop);

    fixture = TestBed.createComponent(DocumentModalComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should display the filter buttons', () => {
    component.currentFilter = ['Billing', 'NewBusiness', 'Endorsement'];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dismiss with selected filter', async () => {
    const modalControllerMock = TestBed.inject(ModalController) as any;
    component.currentFilter = ['Billing', 'NewBusiness', 'Endorsement'];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
    component.onFilterChange({ detail: { value: 'Billing' } });
    const doneButton = fixture.debugElement.query(By.cssAndText('ion-button', 'Done'));
    expect(doneButton).toBeTruthy();
    doneButton.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(modalControllerMock.dismiss).toHaveBeenCalledWith('Billing');
  });

  it('should navigate to the back page when back button is clicked', () => {
    const backButton = fixture.debugElement.query(By.css('csaa-back-button'));
    expect(backButton).toBeTruthy();
  });

  it('should navigate back to documents page when done is clicked', () => {
    const doneButton = fixture.debugElement.query(By.cssAndText('ion-button', 'Done'));
    expect(doneButton).toBeTruthy();
  });
});
