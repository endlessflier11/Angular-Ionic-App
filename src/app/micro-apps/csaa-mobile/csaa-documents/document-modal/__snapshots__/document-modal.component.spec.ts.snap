// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentModalComponent should dismiss with selected filter 1`] = `
<csaa-document-modal
  PolicyDocumentType={[Function Object]}
  __ngContext__={[Function LRootView]}
  configService={[Function ConfigMockService]}
  currentFilter={[Function Array]}
  currentTheme=""
  hasBillingPaymentTypes={[Function Boolean]}
  hasPolicyTypes={[Function Boolean]}
  modalCtrl={[Function AlertControllerMock]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-title>
        Filter
      </ion-title>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Done
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-list>
      <ion-radio-group>
        <ion-item>
          <ion-label>
            All Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="all_documents"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Policy Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="policy_documents"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Billing & Payments
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="billing_&_payments"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Billing Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            New Business
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Endorsements
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
      </ion-radio-group>
    </ion-list>
  </ion-content>
</csaa-document-modal>
`;

exports[`DocumentModalComponent should display the filter buttons 1`] = `
<csaa-document-modal
  PolicyDocumentType={[Function Object]}
  __ngContext__={[Function LRootView]}
  configService={[Function ConfigMockService]}
  currentFilter={[Function Array]}
  currentTheme=""
  hasBillingPaymentTypes={[Function Boolean]}
  hasPolicyTypes={[Function Boolean]}
  modalCtrl={[Function AlertControllerMock]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-title>
        Filter
      </ion-title>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Done
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-list>
      <ion-radio-group>
        <ion-item>
          <ion-label>
            All Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="all_documents"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Policy Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="policy_documents"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Billing & Payments
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
            value="billing_&_payments"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Billing Documents
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            New Business
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
        <ion-item>
          <ion-label>
            Endorsements
          </ion-label>
          <ion-radio
            mode="md"
            slot="end"
          />
        </ion-item>
      </ion-radio-group>
    </ion-list>
  </ion-content>
</csaa-document-modal>
`;
